tgui/packages/tgui/interfaces/PipeDispenser.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2023 @Laboredih123
+ * SPDX-License-Identifier: MIT
+ */
+
+import { useBackend, useLocalState } from '../backend';
+import { Button, Section, Flex, Box, NumberInput, Image } from '../components';
+import { Window } from '../layouts';
+
+export const PipeDispenser = (props, context) => {
+  const { data } = useBackend(context);
+  const disposalpipes = data.disposalpipes || [];
+  const {
+    dispenser_ready,
+    windowName,
+    mobile,
+    removing_pipe,
+    laying_pipe,
+    max_disposal_pipes,
+  } = data;
+  return (
+    <Window
+      title={windowName}
+      width="310"
+      height={mobile ? 360 : 270}>
+      <Window.Content scrollable>
+        <Section>
+          {disposalpipes.map(disposalpipe => {
+            return (
+              <DisposalPipeRow
+                key={disposalpipe.disposaltype}
+                dispenser_ready={dispenser_ready}
+                max_disposal_pipes={max_disposal_pipes}
+                disposalpipe={disposalpipe} />
+            );
+          })}
+        </Section>
+        {!!mobile && (
+          <AutoPipeLaying
+            laying_pipe={laying_pipe}
+            removing_pipe={removing_pipe} />
+        )}
+      </Window.Content>
+    </Window>
+  );
+};
+
+export const DisposalPipeRow = (props, context) => {
+  const { act } = useBackend(context);
+  const [amount, setAmount] = useLocalState(context, 'amount', 1);
+  const {
+    dispenser_ready,
+    max_disposal_pipes,
+    disposalpipe,
+  } = props;
+
+  return (
+    <Flex justify="space-between" align="stretch" style={{ "border-bottom": "1px #555 solid" }}>
+      <Flex.Item>
+        {disposalpipe.image && (
+          <Box style={{ "overflow": "show", "height": "32px" }}>
+            <Image
+              pixelated
+              src={`data:image/png;base64,${disposalpipe.image}`}
+            />
+          </Box>)}
+      </Flex.Item>
+      <Flex.Item
+        grow style={{
+          "display": "flex",
+          "justify-content": "center",
+          "flex-direction": "column",
+        }}>
+        <Box>
+          <Box inline>
+            {disposalpipe.disposaltype}
+          </Box>
+        </Box>
+      </Flex.Item>
```

Comment:
Something's jank with this, I'm not sure specifically what but combining a styled `Flex.Item` with nested `Box`s seems like something's being misused.

Bare minimum I think you can drop the `<Box inline>` layer. If that messes with the vertical alignment then you'd want to tidy up the row as a whole.